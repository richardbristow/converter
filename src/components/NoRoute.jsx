import React from 'react';
import { Link } from 'react-router-dom';

const NoRoute = () => (
  <div>
    Uh-oh! No page exists why dont you go back&nbsp;
    <Link to="/length">home</Link>
    .
  </div>
);

export default NoRoute;
