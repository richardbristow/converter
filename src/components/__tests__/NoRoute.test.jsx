import React from 'react';
import { shallow } from 'enzyme';
import NoRoute from '../NoRoute';

describe('NoRoute', () => {
  it('should render a Error page for a non matched route', () => {
    const wrapper = shallow(<NoRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
