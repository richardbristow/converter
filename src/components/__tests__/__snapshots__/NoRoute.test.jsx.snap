// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoRoute should render a Error page for a non matched route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoRoute />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Uh-oh! No page exists why dont you go back ",
        <Link
          replace={false}
          to="/length"
>
          home
</Link>,
        ".",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Uh-oh! No page exists why dont you go back ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "home",
          "replace": false,
          "to": "/length",
        },
        "ref": null,
        "rendered": "home",
        "type": [Function],
      },
      ".",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Uh-oh! No page exists why dont you go back ",
          <Link
            replace={false}
            to="/length"
>
            home
</Link>,
          ".",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Uh-oh! No page exists why dont you go back ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "home",
            "replace": false,
            "to": "/length",
          },
          "ref": null,
          "rendered": "home",
          "type": [Function],
        },
        ".",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
